### PR TITLE
visual-hexdiff: init at 0.0.53

### PR DESCRIPTION
**Pull request by summersamara**
## Description of changes

Init of hexdiff: Visual hexadecimal difference editor

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}:
+stdenv.mkDerivation {
+  pname = "visual-hexdiff";
+  version = "0.0.53";
+
+  src = fetchurl {
+    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_0.0.53.orig.tar.gz";
+    sha256 = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
+  };
+
```
Comment (erictapen) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:16]:
```suggestion
# Fix compiler error that wants a string literal as format string for `wprintw`
```

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  prePatch = ''
+    substituteInPlace sel_file.c \
+      --replace 'wprintw(win, txt_aide_fs[foo]);' 'wprintw(win, "%s", txt_aide_fs[foo]);'
+  '';
+
+  patches = [
+    (fetchpatch {
```
Comment (erictapen) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:23]:
```suggestion
    # Some changes the debian/ubuntu developers made over the original source code
    (fetchpatch {
```

**Review by erictapen — COMMENTED** [review]
Is there any way to signal the name this package has on Repology? Because currently [they list `hexdiff` as the application introduced here](https://repology.org/project/hexdiff/versions), while Nixpkgs is the only distribution exposing a different hexdiff.

**Comment by FliegendeWurst** [comment]
> Is there any way to signal the name this package has on Repology?

Yes, by making the necessary changes in https://github.com/repology/repology-rules/. Though in this case we may also need to rename hexdiff in nixpkgs.

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  src = fetchurl {
+    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_0.0.53.orig.tar.gz";
+    sha256 = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
+  };
+# Fix compiler error that wants a string literal as format string for `wprintw`
+  prePatch = ''
+    substituteInPlace sel_file.c \
+      --replace 'wprintw(win, txt_aide_fs[foo]);' 'wprintw(win, "%s", txt_aide_fs[foo]);'
+  '';
```
Comment (SuperSandro2000) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:20]:
```suggestion
  postPatch = ''
  	# Fix compiler error that wants a string literal as format string for `wprintw`
    substituteInPlace sel_file.c \
      --replace-fail 'wprintw(win, txt_aide_fs[foo]);' 'wprintw(win, "%s", txt_aide_fs[foo]);'
  '';
```

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  fetchurl,
+  fetchpatch,
+  ncurses,
+}:
+stdenv.mkDerivation {
+  pname = "visual-hexdiff";
+  version = "0.0.53";
+
+  src = fetchurl {
+    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_0.0.53.orig.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:13]:
```suggestion
    url = "mirror://ubuntu/pool/universe/h/hexdiff/hexdiff_${version}.orig.tar.gz";
```

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    sha256 = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
```
Comment (SuperSandro2000) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:14]:
```suggestion
    hash = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
```

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  meta = with lib; {
+    description = "Visual hexadecimal difference editor";
+    homepage = "http://tboudet.free.fr/hexdiff/";
+    license = licenses.wtfpl;
+    maintainers = with maintainers; [ erictapen ];
+    mainProgram = "hexdiff";
+    platforms = platforms.all;
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:50]:
```suggestion
  };
}
```

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+  preInstall = ''
+    mkdir -p $out/bin/
+  '';
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  meta = with lib; {
+    description = "Visual hexadecimal difference editor";
+    homepage = "http://tboudet.free.fr/hexdiff/";
```
Comment (SuperSandro2000) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:43]:
```suggestion
    homepage = "http://tboudet.free.fr/hexdiff/";
```

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    sha256 = "sha256-M1bmkW63pHlfl9zNWEq0EGN1rpVGo+BTUKM9ot4HWqo=";
+  };
+# Fix compiler error that wants a string literal as format string for `wprintw`
+  prePatch = ''
+    substituteInPlace sel_file.c \
+      --replace 'wprintw(win, txt_aide_fs[foo]);' 'wprintw(win, "%s", txt_aide_fs[foo]);'
+  '';
+
+  patches = [
+    # Some changes the debian/ubuntu developers made over the original source code
```
Comment (SuperSandro2000) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:23]: That doesn't explain anything about what the changesare

**Review by SuperSandro2000 — DISMISSED** [review]

File: pkgs/by-name/vi/visual-hexdiff/package.nix (hunk 8 above)
Comment (summersamara) [reply pkgs/by-name/vi/visual-hexdiff/package.nix:23]: added a link to the associated changelog

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  meta = with lib; {
+    description = "Visual hexadecimal difference editor";
+    homepage = "http://tboudet.free.fr/hexdiff/";
+    license = licenses.wtfpl;
+    maintainers = with maintainers; [ erictapen ];
+    mainProgram = "hexdiff";
+    platforms = platforms.all;
```
Comment (FliegendeWurst) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:48]: Please fix or mark as broken for darwin
```
clang -o sel_file.o -c sel_file.c -g -DVERSION=\"0.0.53\" -DTRACE=0  -ansi
sel_file.c:22:7: error: conflicting types for 'strdup'
char *strdup(char *);
      ^
/nix/store/9l0nxr441n8qr0dk5n48m38dz4dswmg5-apple-sdk-10.12.2/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/string.h:117:7: note: previous declaration is here
char    *strdup(const char *__s1);
         ^
1 error generated.
```

Comment (summersamara) [reply pkgs/by-name/vi/visual-hexdiff/package.nix:55]: @FliegendeWurst I pushed a change that might fix the issue

File: pkgs/by-name/vi/visual-hexdiff/package.nix [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
```
Comment (FliegendeWurst) [inline pkgs/by-name/vi/visual-hexdiff/package.nix:55]:
```suggestion
    platforms = platforms.unix;
```
Does not cross-compile for Windows :)

**Review by FliegendeWurst — APPROVED** [review]

**Review by erictapen — APPROVED** [review]

**Pull request merged**